### DISCUSSION TITLE
Plugin on capacitor 2 and ionic 5 ( Error compilation )

### DISCUSSION THREAD

**Discussion by Jrbebel** [timbru31/cordova-plugin-lottie-splashscreen#298] [Q&A] [answered]
Hello @timbru31, 

I tried to install cordova-plugin-splascreen on capacitor2 / ionic5
I have this problem that appears:
![image](https://user-images.githubusercontent.com/3306196/113370545-6bd3fe00-9364-11eb-93e0-4e2e3386eac1.png)

I followed the installation on this link:  https://ionicframework.com/docs/native/lottie-splash-screen

I know that the plugin was not compatible with capacitor at the base but is it possible today?

My config:

![image](https://user-images.githubusercontent.com/3306196/113370715-e0a73800-9364-11eb-935a-eecab5f83688.png)

Thx for help

**Comment by timbru31** [comment]
I think something went wrong earlier. The error message is not from my plugin. _But_ there is a statement "Error adding plugin". Can you post the full verbose build log?

**Reply by timbru31** [reply]
PS: Update your Node.js v12 installation, it contains known vulnerabilities.

**Reply by Jrbebel** [reply]
Hello @timbru31 
i did this command : 

npm install cordova-plugin-lottie-splashscreen
npm install @ionic-native/lottie-splash-screen
ionic cap sync

when i run application in android studio: 
![image](https://user-images.githubusercontent.com/3306196/113528932-5fe57780-95c2-11eb-920f-b022d8acd9d3.png)

**Reply by timbru31** [reply]
Yeah it did not install correctly. This might be due to the fact that build-extras file (https://github.com/timbru31/cordova-plugin-lottie-splashscreen/blob/master/src/android/build-extras.gradle) is not honored. I'll try to reproduce this.

**Reply by timbru31** [reply]
@Jrbebel would you be able to test a git installation?
`npm install https://github.com/timbru31/cordova-plugin-lottie-splashscreen.git#debug`

**Reply by Jrbebel** [reply]
Hello @timbru31, 
I have try but always same errors.
I have try a solution : 

- I convert Kotlin file to Java file ( click rigth on path ../lottie/android/capacitor-cordova-android-plugins/src/main/java/de/dustplanet/cordova/lottie ) 
- I had a problem of version of gradle (version 4 to 6 ), in file gradle-wrapper.properties remplace distributionUrl by distributionUrl=https\://services.gradle.org/distributions/gradle-6.1-all.zip

Now i have this error : 

![image](https://user-images.githubusercontent.com/3306196/114625761-07ddfd80-9cb3-11eb-9f84-e8eb845e5995.png)

If i comment line 284 and line 288, my lottie animation is display 

![image](https://user-images.githubusercontent.com/3306196/114625888-34921500-9cb3-11eb-9ba9-558757539cf6.png)

**Reply by timbru31** [reply]
What's your exact Capacitor version?

**Reply by Jrbebel** [reply]
My version of capacitor is 
Capacitor:
   Capacitor CLI   : 2.4.7
   @capacitor/core : 2.4.7

![image](https://user-images.githubusercontent.com/3306196/114679104-29b79e80-9d0c-11eb-9ac6-ee8d0ea6b7d0.png)

**Reply by timbru31** [reply]
Thanks. I'll need to check their code for the `webView.engine.evaluateJavascript` method

**Comment by timbru31** [comment, marked as answer]
This was fixed by https://github.com/timbru31/cordova-plugin-lottie-splashscreen/pull/438
